pkgs/stdenv/generic/make-derivation.nix

```diff
@@ -9,8 +9,65 @@ let
     # to build it. This is a bit confusing for cross compilation.
     inherit (stdenv) hostPlatform;
   };
+
+  makeOverlayable = mkDerivationSimple: # TODO(@robert): turn mkDerivationSimple into let binding.
+    fnOrAttrs:
+      if builtins.isFunction fnOrAttrs
+      then makeDerivationExtensible mkDerivationSimple fnOrAttrs
+      else makeDerivationExtensible0 mkDerivationSimple fnOrAttrs;
+
+  # Based off lib.makeExtensible, with modifications:
+  makeDerivationExtensible = mkDerivationSimple: rattrs:
```

Comment:
The `mkDerivationSimple` parameter will become a let binding reference, but that (simple) refactoring creates a huge reindentation diff, so I'll do it afterwards.

Reply:
Not worth the hassle and loss of git blame.

Reply:
> loss of git blame.
fyi we can now add revs to `.git-blame-ignore-revs` to ignore them in blame